bndtools.core/src/org/bndtools/core/ui/wizards/service/NewBndServiceWizard.java

```diff
@@ -0,0 +1,195 @@
+	protected String								templateName;
+
+	protected TemplateParamsWizardPage createParamsPage() {
+		return new TemplateParamsWizardPage(ProjectTemplateParam.valueStrings());
+	}
+
+	protected NewBndServiceWizardNonApiProjectPage createImplPage() {
+		return new NewBndServiceWizardNonApiProjectPage(
+			new NewBndServiceWizardPageOne(pageOne, ServiceTemplateConstants.DEFAULT_IMPL_SUFFIX, templateName));
+	}
+
+	protected NewBndServiceWizardNonApiProjectPage createConsumerPage() {
+		return new NewBndServiceWizardNonApiProjectPage(
+			new NewBndServiceWizardPageOne(pageOne, ServiceTemplateConstants.DEFAULT_CONSUMER_SUFFIX, templateName));
+	}
+
+	public NewBndServiceWizard(final NewBndServiceWizardPageOne pageOne, final NewJavaProjectWizardPageTwo pageTwo,
+		String templateName) {
+		super(pageOne, pageTwo);
+		this.pageOne = pageOne;
+		this.templateName = templateName;
+		// setup params page with normal bnd template project params
+		this.paramsPage = createParamsPage();
+		// setup impl page to use .impl project suffix
+		this.implPage = createImplPage();
+		this.consumerPage = createConsumerPage();
+	}
+
+	@SuppressWarnings("restriction")
+	@Override
+	protected void finishPage(IProgressMonitor monitor) throws InterruptedException, CoreException {
+		try {
+			SubMonitor submonitor = SubMonitor.convert(monitor, "Creating service projects", 3);
+			super.finishPage(submonitor.split(1));
+			implPage.performFinish(submonitor.split(1));
+			consumerPage.performFinish(submonitor.split(1));
+		} finally {
+			monitor.done();
+		}
+	}
+
+	@Override
+	public void addPages() {
+		addPage(pageOne);
+		addPage(paramsPage);
+	}
+
+	@Override
+	protected Map<String, String> getServiceTemplateParams() {
+		Map<String, String> params = new HashMap<>();
+		// Service Name
+		params.put(ServiceProjectTemplateParam.SERVICE_NAME.getString(), pageOne.getServiceName());
+		// Package Name
+		String packageName = pageOne.getPackageName();
+		params.put(ProjectTemplateParam.BASE_PACKAGE_NAME.getString(), packageName);
+		// api_package this is the packageName of the API package, made
+		// available
+		// for impl and consumer templates so that the API package can be
+		// imported
+		params.put(ServiceProjectTemplateParam.API_PACKAGE.getString(), packageName);
+		// Package Dir
+		String packageDir = packageName.replace('.', '/');
+		params.put(ProjectTemplateParam.BASE_PACKAGE_DIR.getString(), packageDir);
+		// Version
+		params.put(ProjectTemplateParam.VERSION.getString(), DEFAULT_BUNDLE_VERSION);
+		// Source Folders
+		IJavaProject javaProject = pageTwo.getJavaProject();
+		Map<String, String> sourceOutputLocations = JavaProjectUtils.getSourceOutputLocations(javaProject);
+		int nr = 1;
+		for (Map.Entry<String, String> entry : sourceOutputLocations.entrySet()) {
+			String src = entry.getKey();
+			String bin = entry.getValue();
+			if (nr == 1) {
+				params.put(ProjectTemplateParam.SRC_DIR.getString(), src);
```

Comment:
Are you sure to make these come from the workspace?

Reply:
Same as above...i.e. which workspace?   Also...can't the second java page be used to set these for project creation (admittedlyI I'm not showing that page under typical usage, but it can be made to appear/set for created project(s).

Reply:
Hmm, there is always a Workspace. You can get it from Central or as a Workspace service. There is only going to be one Workspace object in bndtools. If the workspace changes, we violate the existing singleton to reference the new directories. So only need to get it once. If you're really cool, you can get events when they get updated, see the `on(name)` method.

The `src`, `test`, `target`, etc. macros refer to the relative paths in a project where the working directories reside.

Reply:
Right...the relative paths *in a project* (not all projects currently in workspace).  I took this code from the Bndtools project wizards FWIW...it's not like I'm inventing a new approach for bndtools by doing things this way.   I'm just copying what's there in bndtools own project creation wizard....mostly so that it's visually and behaviorally familiar to as many users as possible.

As a proposed compromise, I'll remove bundle-version and leave javaLevel and paths...since these are set at the project level in the current project creation UIs.